types/src/vm/deserialization.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright © Aptos Foundation
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::vm::environment::AptosEnvironment;
+use move_binary_format::{
+    deserializer::DeserializerConfig, errors::PartialVMError, CompiledModule,
+};
+
+pub trait Deserializable: Sized {
```

Comment:
Do we need to introduce this new trait? Seems to me a bit of an overkill as we always need to use the config?

Reply:
Block-STM task is generic, and hides arguments to initialize the VM under an associated type. Do you think making it concrete is better? If so, I will be able to propagate it to `LatestView` as is, without using these traits. I don't have any preference here

Reply:
I see, you actually talk about `Deserializable`! Probably not in this PR.

I think with these traits (as per design doc) we can actually implement MoveVM - adapter interfaces relatively generically, as the code needs to
-> be deserializable
-> be verifiable
-> allow to fetch all transitive or just immediate dependencies
-> (maybe) have metadata
With these we do not care if it is a `CompiledModule`, a `Module`, or some other format... I wrote about that in the design doc.

Let's start gradually though, I ll remove them for now and we can add them if there are really needed later!

Reply:
@runtian-zhou updated trait to fetch config instead, this way it is not an overkill and is sufficient for what we have